File: AlgorithmsPackCPP/Sqrt(x).cpp
class Solution {
public:
    double dubs(int x, double EPS = 1e-9){
        double start = 0.0;
        double end = x;
        while(end-start>EPS){
            double mid = start+(end-start)/2.0;
            if(mid*mid-x<0.0)
                start = mid;
            else
                end = mid;
        }
        return start+EPS;
    }

    int mySqrt(int x) {
        return dubs(x);
    }
};

File: AlgorithmsPackJavaSolutions/Sqrt(x).java
class Solution {
    public int mySqrt(int x) {
        double start = 0.0;
        double end = (double) x;
        double EPS = 1e-9;
        while (end - start > EPS) {
            double mid = start + (end - start) / 2.0;
            if (mid * mid - x < 0.0)
                start = mid;
            else
                end = mid;
        }
        return (int) (start + EPS);
    }
}
